source/distributed/tria.cc

```diff
@@ -3122,6 +3122,237 @@ namespace parallel
 
 
 
+    namespace
+    {
+      template <typename ITERATOR>
+      void
+      identify_periodic_vertices_recursively(const GridTools::PeriodicFacePair<ITERATOR> &periodic,
+                                             std::vector<unsigned int> &topological_vertex_numbering)
```

Comment:
I don't think I quite understand what `topologic_vertex_numbering` is supposed to be. Can you add a comment to the function?